config/merge_config.go

```diff
@@ -0,0 +1,377 @@
+	}
+
+	agentConf.Scrubber.Enabled = mergeBool(dc, keyScrubArgs, agentConf.Scrubber.Enabled)
+
+	csw := mergeString(dc, keyCustomSensitiveWords, "")
+	agentConf.Scrubber.AddCustomSensitiveWords(strings.Split(csw, ","))
+	agentConf.Scrubber.StripAllArguments = mergeBool(dc, keyStripProcessArguments, agentConf.Scrubber.StripAllArguments)
+
+	agentConf.QueueSize = mergeInt(dc, keyQueueSize, agentConf.QueueSize)
+
+	agentConf.MaxProcFDs = mergeInt(dc, keyMaxProcFDs, agentConf.MaxProcFDs)
+
+	if mpm := mergeInt(dc, keyMaxPerMessage, agentConf.MaxPerMessage); mpm <= maxMessageBatch {
+		agentConf.MaxPerMessage = mpm
+	} else {
+		log.Warn("Overriding the configured item count per message limit because it exceeds maximum")
+	}
+
+	agentConf.DDAgentBin = mergeString(dc, keyDDAgentBin, agentConf.DDAgentBin)
+
+	agentConf.DDAgentPy = mergeString(dc, keyDDAgentPy, agentConf.DDAgentPy)
+
+	agentConf.DDAgentPyEnv = strings.Split(mergeString(dc, keyDDAgentPyEnv, ""), ",")
+
+	winAri := mergeInt(dc, keyWinArgsRefreshInterval, agentConf.Windows.ArgsRefreshInterval)
+	if winAri != 0 {
+		agentConf.Windows.ArgsRefreshInterval = winAri
+	}
+
+	agentConf.Windows.AddNewArgs = mergeBool(dc, keyWinAddNewArgs, agentConf.Windows.AddNewArgs)
+
+	if dc.IsSet(keyAdditionalEndpoints) {
+		additionalEndpoints := dc.GetStringMapStringSlice(keyAdditionalEndpoints)
+		for endpointURL, apiKeys := range additionalEndpoints {
+			u, err := url.Parse(endpointURL)
+			if err != nil {
+				return fmt.Errorf("invalid additional endpoint url '%s': %s", endpointURL, err)
+			}
+			for _, k := range apiKeys {
+				agentConf.APIEndpoints = append(agentConf.APIEndpoints, APIEndpoint{
+					APIKey:   k,
+					Endpoint: u,
+				})
+			}
+		}
+	}
+
+	// Pull additional parameters from the global config file.
+	agentConf.LogLevel = mergeString(dc, "log_level", agentConf.LogLevel)
+	agentConf.StatsdPort = mergeInt(dc, "dogstatsd_port", agentConf.StatsdPort)
+	agentConf.StatsdHost = mergeString(dc, "bind_host", agentConf.StatsdHost)
+	agentConf.Transport = ddutil.CreateHTTPTransport()
+
+	// Network related config
+	if ok, _ := isAffirmative(dc.GetString(keyNetworkTracingEnabled)); ok {
+		agentConf.EnabledChecks = append(agentConf.EnabledChecks, "connections")
+		agentConf.EnableNetworkTracing = true
+	}
+
+	agentConf.NetworkTracerSocketPath = mergeString(dc, keyNetworkUnixSocketPath, agentConf.NetworkTracerSocketPath)
+	agentConf.NetworkTracerLogFile = mergeString(dc, keyNetworkLogFile, agentConf.NetworkTracerLogFile)
+
+	agentConf.DisableTCPTracing, _ = isAffirmative(dc.GetString(keyNetworkDisableTCPTracing))
+	agentConf.DisableUDPTracing, _ = isAffirmative(dc.GetString(keyNetworkDisableUDPTracing))
+	agentConf.DisableIPv6Tracing, _ = isAffirmative(dc.GetString(keyNetworkDisableIPV6Tracing))
+
+	return nil
+}
+
+// mergeEnvironmentVariables applies overrides from environment variables to the process agent configuration
+func mergeEnvironmentVariables(dc ddconfig.Config, c *AgentConfig) {
```

Comment:
nit: if we can not avoid reading API_KEy and LOG_LEVEL in this function, maybe call this mergeEnvironmentOnlyConfig as this handles all the things that are *not* configured in configuration files. `mergeEnvironmentVariables` feels wrong because `mergeConfig` will read environment variables if it finds them.